### PR TITLE
network: fix IPv4 packets being treated as IPv6 (net_packet_ipv6 only traced)

### PR DESCRIPTION
**Pull request by rafaeldtinoco**
network: fix retval for IPv6 packets

The event retval value works, for networking events, as a value to
tell userland (derivation) tips about what to do. Initial usecase was
to differentiate layer 3 protocols (IPv4 vs IPv6), but it gained more
meaning with http events.

This patch fixes the retval value for IPv6 packets, fixing issues seen
with net_packet_ipv6 events having zeroed values.

### CODE REVIEW

File: pkg/ebpf/c/tracee.bpf.c [hunk 1]
```diff
@@ -4285,12 +4285,10 @@ int BPF_KPROBE(cgroup_bpf_run_filter_skb)
     eventctx->stack_id = 0;                                 // no stack trace
     eventctx->processor_id = p.event->context.processor_id; // copy from current ctx
     eventctx->matched_scopes = netctx->matched_scopes;      // pick matched-scopes from net ctx
-    if (type == BPF_CGROUP_INET_EGRESS) {
-        eventctx->syscall = netctx->syscall;
-    } else {
-        // During ingress there is no real originated syscall
-        eventctx->syscall = NO_SYSCALL;
-    }
+    eventctx->syscall = NO_SYSCALL;                         // ingress has no orig syscall
```
Comment (rafaeldtinoco) [inline pkg/ebpf/c/tracee.bpf.c:4288]: this is just an opportunity for style adjustment.

File: pkg/ebpf/c/tracee.bpf.c [hunk 2]
```diff
@@ -4601,11 +4599,6 @@ CGROUP_SKB_HANDLE_FUNCTION(family)
 
     neteventctx->md.header_size = size; // add header size to offset
 
-    // submit the IP base event
```
Comment (rafaeldtinoco) [inline pkg/ebpf/c/tracee.bpf.c:4604]: need to process IPv6 header before submitting the network packet to userland.

File: pkg/ebpf/c/tracee.bpf.c [hunk 3]
```diff
@@ -4661,6 +4654,30 @@ CGROUP_SKB_HANDLE_FUNCTION(proto)
...
+                case 254:                   // experimentation and testing
+                    bpf_printk("IPv6 extension header unsupported");
+                    return 1;
+            }
+
+            bpf_printk("IPv6 extension header: %d", next_proto);
+
+            // submit the IP base event
+
+            if (should_submit_net_event(neteventctx, SUB_NET_PACKET_IP))
```
Comment (rafaeldtinoco) [inline pkg/ebpf/c/tracee.bpf.c:4678]: now it can be submitted.

File: pkg/ebpf/c/missing_definitions.h [hunk 4]
```diff
@@ -4,88 +4,88 @@
 // NOTE: vmlinux.h contains kernel BTF information but not macros.
 
 #ifndef __TRACEE_MISSING_MACROS_H__
-#define __TRACEE_MISSING_MACROS_H__
```
Comment (rafaeldtinoco) [inline pkg/ebpf/c/missing_definitions.h:7]: With latest changes, fix-fmt should be applied to all .c and .h files on the tree. This also fixes issues making the PR not to pass the check-fmt command.

**Comment by rafaeldtinoco** [comment]
Still working on this one.

File: builder/Makefile.checkers [hunk 5]
```diff
@@ -74,17 +74,19 @@ help:
 # check formatting (clang-format, gofmt)
 #
 
+C_FILES_TO_BE_CHECKED = $(shell find ./pkg/ebpf/c/ -regextype posix-extended -regex '.*\.(h|c)' | xargs)
```
Comment (NDStrahilevitz) [inline builder/Makefile.checkers:77]: Thanks for this.

**Review by geyslan — APPROVED** [review]
LGTM

**Pull request merged**